bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,360 @@
+    WasmDocument::from(document)
+  }
+
+  /// Resolves the DID Document associated with this `Account` from the Tangle.
+  #[wasm_bindgen(js_name = resolveIdentity)]
+  pub fn resolve_identity(&self) -> PromiseResolvedDocument {
+    let account = self.0.clone();
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow()
+        .resolve_identity()
+        .await
+        .map(WasmResolvedDocument::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+    promise.unchecked_into::<PromiseResolvedDocument>()
+  }
+
+  /// Removes the identity from the local storage entirely.
+  ///
+  /// Note: This will remove all associated document updates and key material - recovery is NOT POSSIBLE!
+  #[wasm_bindgen(js_name = deleteIdentity)]
+  pub fn delete_identity(self) -> Promise {
+    //ToDo: test once `load_identity(..)` is implemented.
+
+    // Get IotaDID and storage from the account.
+    let account: Rc<WasmRefCell<Account>> = self.0;
+    let did: IotaDID = account.as_ref().borrow().did().to_owned();
+    let storage: Arc<dyn Storage> = account.as_ref().borrow().storage_arc();
+
+    // Drop account should release the DIDLease because we cannot take ownership of the Rc.
+    // Note that this will still fail if anyone else has a reference to the Account.
+    std::mem::drop(account);
+
+    future_to_promise(async move {
+      // Create a new account since `delete_identity` consumes it.
+      let account: Result<Account> = AccountBuilder::new()
+        .storage(AccountStorage::Custom(storage))
+        .load_identity(did)
+        .await
+        .wasm_result();
+
+      match account {
+        Ok(a) => a.delete_identity().await.wasm_result().map(|_| JsValue::undefined()),
+        Err(e) => Err(e),
+      }
+    })
+  }
+
+  /// Push all unpublished changes to the tangle in a single message.
+  #[wasm_bindgen]
+  pub fn publish(&mut self, publish_options: Option<WasmPublishOptions>) -> Promise {
+    let account = self.0.clone();
+    if let Some(publish_options) = publish_options {
+      let mut options: PublishOptions = PublishOptions::new();
+
+      if let Some(force_integration) = publish_options.forceIntegrationUpdate() {
+        options = options.force_integration_update(force_integration);
+      }
+
+      if let Some(sign_with) = publish_options.signWith() {
+        let s: String = sign_with;
+        options = options.sign_with(s);
+      };
+      future_to_promise(async move {
+        account
+          .as_ref()
+          .borrow_mut()
+          .publish_with_options(options)
+          .await
+          .map(|_| JsValue::undefined())
+          .wasm_result()
+      })
+    } else {
+      future_to_promise(async move {
+        account
+          .as_ref()
+          .borrow_mut()
+          .publish()
+          .await
+          .map(|_| JsValue::undefined())
+          .wasm_result()
+      })
+    }
```

Comment:
```suggestion
    let mut options: PublishOptions = PublishOptions::default();

    if let Some(publish_options) = publish_options {
      if let Some(force_integration) = publish_options.forceIntegrationUpdate() {
        options = options.force_integration_update(force_integration);
      }

      if let Some(sign_with) = publish_options.signWith() {
        options = options.sign_with(sign_with);
      };
    };

    future_to_promise(async move {
      account
        .as_ref()
        .borrow_mut()
        .publish_with_options(options)
        .await
        .map(|_| JsValue::undefined())
        .wasm_result()
    })
```

That achieves the same :slightly_smiling_face: `Account::publish` is the same as `publish_with_options` with `PublishOptions::default()`.